e2e/pagination/pagination.e2e-spec.ts

```diff
@@ -26,44 +55,63 @@ test.describe.parallel(`Pagination tests`, () => {
 	const initState: State = {
 		rootClasses: ['au-pagination', 'pagination'],
 		disabled: null,
-		pages: ['1(current)'],
+		isPreviousDisabled: false,
+		isNextDisabled: false,
+		pages: ['1', '2', '3', '4(current)', '5', '6', '7'],
+	};
+	const initCustomState: State = {
+		rootClasses: ['au-pagination', 'pagination'],
+		disabled: null,
+		isPreviousDisabled: false,
+		isNextDisabled: false,
+		pages: ['A', 'B', 'C', 'D(current)', 'E', 'F', 'G'],
+	};
+	const disabledInitState: State = {
+		rootClasses: ['au-pagination', 'pagination'],
+		disabled: 'true',
+		isPreviousDisabled: true,
+		isNextDisabled: true,
+		pages: ['1(current)', '2', '3', '4', '5', '6', '7'],
 	};
 	test(`Default features`, async ({page}) => {
 		const paginationDemoPO = new PaginationDemoPO(page);
 		const paginationPO = new PaginationPO(page, 0);
-
 		await page.goto('#/pagination/default');
 		await paginationPO.locatorRoot.waitFor();
-
 		expect(await paginationState(paginationPO)).toEqual(initState);
+		expect(await paginationDemoPO.defaultPaginationDemoState()).toEqual({page: 4});
+		await expect(paginationPO.locatorPreviousButton).not.toBeDisabled();
+		await expect(paginationPO.locatorNextButton).not.toBeDisabled();
+		const disabledPaginationPO = new PaginationPO(page, 3);
+		expect(await paginationState(disabledPaginationPO)).toEqual(disabledInitState);
+		const paginationWithBoundariesPO = new PaginationPO(page, 2);
+		await paginationWithBoundariesPO.locatorFirstButton.click();
 		expect(await paginationDemoPO.defaultPaginationDemoState()).toEqual({page: 1});
-
-		await expect(paginationPO.locatorPreviousButton).toBeDisabled();
-		await expect(paginationPO.locatorNextButton).toBeDisabled();
+		expect(await paginationState(paginationWithBoundariesPO)).toEqual({
+			...initState,
+			isFirstDisabled: true,
+			isLastDisabled: false,
+			isPreviousDisabled: true,
+			pages: ['1(current)', '2', '3', '4', '5', '6', '7'],
+		});
+		await paginationWithBoundariesPO.locatorLastButton.click();
+		expect(await paginationDemoPO.defaultPaginationDemoState()).toEqual({page: 7});
+		expect(await paginationState(paginationWithBoundariesPO)).toEqual({
+			...initState,
+			isFirstDisabled: false,
+			isLastDisabled: true,
+			isNextDisabled: true,
+			isPreviousDisabled: false,
+			pages: ['1', '2', '3', '4', '5', '6', '7(current)'],
+		});
 	});
 
 	// TODO add test with the custom template, className...
-	test(`Config features`, async ({page}) => {
-		let expectedState = {...initState};
-		const paginationDemoPO = new PaginationDemoPO(page);
+	test(`Custom features`, async ({page}) => {
+		const expectedState = {...initCustomState};
 		const paginationPO = new PaginationPO(page, 0);
-
-		await page.goto('#/pagination/config');
+		await page.goto('#/pagination/custom');
 		await paginationPO.locatorRoot.waitFor();
-
```

Comment:
Will you add more testing to the custom route in this PR ?
If not please open a GitHub issue so that we do not forget to implement testing on the custom route.

Reply:
done